src/media_type.rs

```diff
@@ -262,24 +283,6 @@ impl fmt::Display for MediaType {
   }
 }
 
-impl<'a> From<&'a Path> for MediaType {
-  fn from(path: &'a Path) -> Self {
-    Self::from_path(path)
-  }
-}
-
-impl<'a> From<&'a PathBuf> for MediaType {
-  fn from(path: &'a PathBuf) -> Self {
-    Self::from_path(path)
-  }
-}
-
-impl<'a> From<&'a String> for MediaType {
-  fn from(specifier: &'a String) -> Self {
-    Self::from_path(&PathBuf::from(specifier))
-  }
-}
```

Comment:
I also got rid of these `From` implementations because they are not discoverable. `From` implementations are good in some cases, but not good as a replacement for `new_` methods because they're hard to know about. With `new_` methods I get auto complete.

Reply:
Strong agree